galeracluster/source/installmariadbsrc.rst

```diff
@@ -5,61 +5,189 @@ Source Installation
 
-For system uses an RPM-based distribution, instead run this command:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Building the Database Server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _`build-mariadb`:
+
+The database server for Galera Cluster is the same as that of the standard database servers for  standalone instances of MariaDB, with the addition of a patch for the wsrep API, which is packaged in the version downloaded from `GitHub <https://github.com>`_.  You can enable the patch through the ``WITH_WSREP`` and ``WITH_INNODB_DISALLOW_WRITES`` CMake configuration options.
+
+To build the database server, run the following commands from the ``mariadb/`` directory:
 
 .. code-block:: console
 
-   # yum-builddep MariaDB-server
+   # cmake -DWITH_WSREP=ON -DWITH_INNODB_DISALLOW_WRITES=ON ./
+   # make
+   # make install
+
+
+.. note:: In addition to compiling through ``cmake`` and ``make``, there are also a number of build scripts in the ``BUILD/`` directory, which you may find more convenient to use.  For example, 
 
-For other distributions and unix-like operating systems, consult the documentation for the appropriate package manager and syntax.
+	  .. code-block:: console
 
+	     # ./BUILD/compile-pentium64-max
 
---------------------------------------------
-Building the Galera Cluster for MariaDB
---------------------------------------------
-.. _`Build Galera MariaDB`:
+	  This has the same effect as running the above commands with various build options pre-configured.  There are several build scripts available in the directory, select the one that best suits your needs.
 
-The source code for MariaDB Galera Cluster is available through Launchpad.net.  Using Bazaar with the ``branch`` argument, you can download the source code for your specific system:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Building the wsrep Provider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. _`build-mariadb-galera`:
+
+The :term:`Galera Replication Plugin` implements the :term:`wsrep API` and operates as the wsrep Provider for the database server.  What it provides is a certification layer to prepare write-sets and perform certification checks, a replication layer and a group communication framework.  
+
+To build the Galera Replication Plugin, run SCons from the ``galera/`` directory.
 
 .. code-block:: console
 
-   $ bzr branch lp:maria trunk
+   # scons
 
-Once Bazaar finishes running, you can start building the database server.  You have two options for how to build MariaDB Galera Cluster.  You can use a build script or you can build it using ``cmake``.  If you choose to use a build script, the command will vary depending upon your system architecture.
+This process creates the Galera Replication Pluigin, (that is, the ``libgalera_smm.so`` file).  In your configuration file, you need to define the path to this file for the :ref:`wsrep_provider <wsrep_provider>` parameter.
```

Comment:
in your my.cnf configuration file, you need to specify the path to this file in the wsrep_provider_parameter ...